pkg/kubelet/util/queue/queue.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package queue
+
+import (
+	"sync"
+)
+
+// Queue allows queuing items
+type Queue interface {
```

Comment:
Do we need this abstract type?  It would be much easier to use an existing container type or just a simple synced map.  I don't know that order of deletion matters?

Reply:
I only want to use the queue's advantage that items have orders and does not need a remove while calling get(), I'll think your suggestions too.